apps/setting/models_provider/impl/gemini_model_provider/gemini_model_provider.py

```diff
@@ -56,14 +59,23 @@
               GeminiSpeechToText),
 ]
 
+model_embedding_info_list = [
+    ModelInfo('models/embedding-001', '',
+              ModelTypeConst.EMBEDDING,
+              gemini_embedding_model_credential,
+              GeminiEmbeddingModel),
+]
+
 model_info_manage = (
     ModelInfoManage.builder()
     .append_model_info_list(model_info_list)
     .append_model_info_list(model_image_info_list)
     .append_model_info_list(model_stt_info_list)
+    .append_model_info_list(model_embedding_info_list)
     .append_default_model_info(model_info_list[0])
     .append_default_model_info(model_image_info_list[0])
     .append_default_model_info(model_stt_info_list[0])
+    .append_default_model_info(model_embedding_info_list[0])
     .build()
 )
 
```

Comment:
Here's a checklist of issues and improvements:

### Issues:

1. **Duplicate `build()` call**: There seems to be an extra `build()` method invocation inside another block.

2. **Redundant credentials and models**: The same credentials (`gemini_llm_model_credential`, etc.) are reused for multiple types (`GeminiChatModel`, `GeminiSpeechToText`), which might lead to confusion or errors if additional configurations are needed for different types.

3. **Missing import statements**: Ensure that all necessary classes are imported correctly at the top.

4. **Inconsistent indentation**: Check that all indents are consistent throughout the file.

5. **Logical consistency**: Verify that the order of model information addition is logical (e.g., LLMs before images) as this could affect how settings load.

### Optimizations and Improvements:

1. **Single `Credential` Instance per Model**: If each type has distinct credentials, consider merging them into a single instance with appropriate differentiation logic.

2. **Separate Credential Classes**: If possible, use separate credential classes tailored specifically to each model type or add dynamic behavior through class methods.

3. **Remove Unused Code Blocks**: Review redundant blocks like the initial comment at -11, and remove unused imports.

4. **Consistent Error Handling**: Implement proper error handling mechanisms to manage missing dependencies or invalid data.

5. **Testing Coverage**: Add tests to ensure that all functionalities work as expected after making changes.

6. **Refactor Method Calls**: Simplify complex calls by breaking down operations where appropriate.

7. **Documentation**: Update comments and docstrings to describe the purpose and functionality of code components.

8. **Version Control**: Use version control to track changes made during reviews.

By addressing these items, you can enhance the readability, maintainability, and overall quality of your codebase.